sdgym/utils.py

```diff
@@ -47,19 +48,31 @@ def get_synthesizers(synthesizers):
         TypeError:
             If neither a list is not passed.
     """
+    st_sdv_synthesizers = _get_sdv_synthesizers('single_table')
+    mt_sdv_synthesizers = _get_sdv_synthesizers('multi_table')
     synthesizers = [] if synthesizers is None else synthesizers
     if not isinstance(synthesizers, list):
         raise TypeError('`synthesizers` must be a list.')
 
     synthesizers_dicts = []
+    baselines = BaselineSynthesizer.get_subclasses(include_parents=True)
     for synthesizer in synthesizers:
         if isinstance(synthesizer, str):
-            baselines = BaselineSynthesizer.get_subclasses(include_parents=True)
-            if synthesizer in baselines:
-                LOGGER.info('Trying to import synthesizer by name.')
-                synthesizer = baselines[synthesizer]
+            LOGGER.info('Trying to import synthesizer by name.')
+            synthesizer_name = synthesizer
+            if synthesizer in st_sdv_synthesizers + mt_sdv_synthesizers:
+                modality = 'single_table' if synthesizer in st_sdv_synthesizers else 'multi_table'
+                instance = BaselineSDVSynthesizer(synthesizer, modality)
```

Comment:
Instead of directly returning an instance of BaselineSDVSynthesizer, can you add a function, `create_sdv_synthesizer_class` and in it return something like:
```python
    CustomSynthesizer = type(
        class_name, # Should be the synthesizer name in sdv
        (BaselineSynthesizer,),
        {
            '__module__': __name__,
            'get_trained_synthesizer': get_trained_synthesizer,
            'sample_from_synthesizer': sample_from_synthesizer,
        },
    )
```